src/notebook.c

```diff
@@ -531,6 +550,8 @@ void notebook_init(void)
 	/* in case the switch dialog misses an event while drawing the dialog */
 	g_signal_connect(main_widgets.window, "key-release-event", G_CALLBACK(on_key_release_event), NULL);
 
+	g_signal_connect(geany_object, "document-open", G_CALLBACK(on_document_open), NULL);
```

Comment:
you probably also need to catch `document-save` in case the document has been saved as.